jwt.go

```diff
@@ -241,19 +300,33 @@ func (ja *JWTAuth) validateSignatureKeys() error {
 	return nil
 }
 
-func (ja *JWTAuth) keyProvider() jws.KeyProviderFunc {
-	return func(_ context.Context, sink jws.KeySink, sig *jws.Signature, _ *jws.Message) error {
+func (ja *JWTAuth) keyProvider(request *http.Request) jws.KeyProviderFunc {
+	return func(context context.Context, sink jws.KeySink, sig *jws.Signature, _ *jws.Message) error {
 		if ja.usingJWK() {
+			// Resolve JWKURL with placeholders
+			replacer := request.Context().Value(caddy.ReplacerCtxKey).(*caddy.Replacer)
```

Comment:
Even there're only two line of code here, I would prefer creating a separate function to get the JWK URL which is bound to the given HTTP request.